### ISSUE TITLE
[BUG]: App Always shows the connection Username not the Logged in User

### ISSUE THREAD

**Issue by Lancelot1976** [microsoft/powerapps-tools#169]
I have imported the Crisis Communication App and replaced the Office365Users Connection with a systemuser for accessing the Object. Now when I log into the app with my User Account, it Always show the systemuser Username and all actions are connected with this User e.g. Work status

What I am doing wrong?

**Comment by audrieMSFT** [comment]
Could you explain what you did here in more detail, and the purpose?
"...and replaced the Office365Users Connection with a systemuser for accessing the Object"

**Comment by Lancelot1976** [comment]
Yes, of course ;-) On the import I have to select a connection for the Office 365 Users Object. There I select a User for the connection nd this user is now always the one who is shown...

![31-import-app](https://user-images.githubusercontent.com/62646752/77529978-eb490a80-6e90-11ea-95ad-50ebda4972b6.png)

**Comment by PaulieShore** [comment]
We are seeing the same behavior. For example, when we use the "Update Status" option, the email notification is delivered from the Service Account, not the User Account in which completed the request. I'm curious if this might be related to Exchange Hybrid configurations. When I tried to create the Office 265 Outlook connection, I had to specific an account which had a mailbox within Exchange Online (I received an error when selecting a user account with on-prem mailbox). Perhaps this is related to our issue?

**Comment by Lancelot1976** [comment]
We also use the Exchange Hybrid configuration... and we have the same behavior. With the Outlook Connection I understand the behavior, because for sending the mail he uses the mailbox of the service account and have no access to the user mailbo. But the Office 365 Users connection should access the login user data and not the connection user data. When I look in the code, he uses Otulook365Users.MyProfile_V2 function and this returns the connection user not the login user, if I understand the documentation correctly.... But then everyone should have this issue

**Comment by Lancelot1976** [comment]
I have replaced the occurence of Office365Users.MyProfileV2() with Office365Users.UserProfileV2(User().Email) 

This seems to work

**Comment by PaulieShore** [comment]
> I have replaced the occurence of Office365Users.MyProfileV2() with Office365Users.UserProfileV2(User().Email)
> 
> This seems to work

I now understand what you're referring to. I had originally deployed the apps as my own user account, which is why I never noticed the issues with the actual app showing the Service Account rather than my account. Once I re-deployed the app using a proper Service Account, I can see the problem. 

Would you be able to provide a quick overview of where you had to change Office365Users.MyProfileV2() to Office365Users.UserProfileV2(User().Email)? Many thanks.

**Comment by Lancelot1976** [comment]
Yes, I have done the same in the beginning. After changing only the account it did not work and I tried to redeploy the app with the service account then, but the same issue. So it seems to be regardless if you deploy it or change it afterwards.

MY change is only a quick fix and not working completely at the moment. The Presence Status do not work for now.

I change it at the moment in 

Home Screen -> Extended -> Action On Visible 

and 

HomeComponent -> Extended -> Data User

I hope the translations are correct, because I use the german UI ;-)

Now it seems that the collection colDates also refers to the wrong user, but I didn't find how to change it for now...

**Comment by PaulieShore** [comment]
So I guess the question remains: Why are we seeing this behavior? I would think it would impact everyone. It almost seems as if the accounts were using for the Connectors don't have the correct permissions. Either that or the app is relying heavily on the Office 365 Outlook connector, in which case may not work for those of us with Exchange Hybrid and on-prem mailboxes.

**Comment by Lancelot1976** [comment]
Yes you are rigth...  Maybe we have a wrong version downloaded. By the way I found the collection update:

Home Screen -> Extended -> Action On Visible 

at the end uncomment

// Get all dates of the logged-in user
    **ClearCollect(colDates;Filter('CI_Employee Status';Title=varUser.userPrincipalName));**

Maybe it depends on the Hybrid configuration. I don't know. I started with Power Apps on Monday, so I am not so experienced in this technology... 

For today I will be off, maybe we have new infos tommorrow....

**Comment by PaulieShore** [comment]
> Yes you are rigth... Maybe we have a wrong version downloaded. By the way I found the collection update:
> 
> Home Screen -> Extended -> Action On Visible
> 
> at the end uncomment
> 
> // Get all dates of the logged-in user
> **ClearCollect(colDates;Filter('CI_Employee Status';Title=varUser.userPrincipalName));**
> 
> Maybe it depends on the Hybrid configuration. I don't know. I started with Power Apps on Monday, so I am not so experienced in this technology...
> 
> For today I will be off, maybe we have new infos tommorrow....

I'm new to PowerApps as well. Just started today. Thanks for all the info. It's greatly appreciated. Hopefully we'll see an update this evening.

**Comment by Lancelot1976** [comment]
I think it is a problem with the hybrid installation. All local users can't use the app, because the Outlook account is not accessible in Office 365

**Comment by PaulieShore** [comment]
I guess the app just won't work for those of us using a hybrid configuration.
